internal/distro/fedora31/distro.go

```diff
@@ -16,45 +15,45 @@ import (
 	"github.com/osbuild/osbuild-composer/internal/rpmmd"
 )
 
-type arch struct {
-	Name               string
-	BootloaderPackages []string
-	BuildPackages      []string
-	UEFI               bool
-}
-
-type output struct {
-	Name             string
-	MimeType         string
-	Packages         []string
-	ExcludedPackages []string
-	EnabledServices  []string
-	DisabledServices []string
-	KernelOptions    string
-	Bootable         bool
-	DefaultSize      uint64
-	Assembler        func(uefi bool, size uint64) *osbuild.Assembler
-}
-
-const Distro = common.Fedora31
-const ModulePlatformID = "platform:f31"
+const name = "fedora-31"
+const modulePlatformID = "platform:f31"
 
 type Fedora31 struct {
 	arches        map[string]arch
-	outputs       map[string]output
+	imageTypes    map[string]imageType
 	buildPackages []string
 }
 
-type Fedora31Arch struct {
+type arch struct {
+	name               string
+	bootloaderPackages []string
+	buildPackages      []string
+	uefi               bool
+}
+
+type imageType struct {
+	name             string
+	mimeType         string
+	packages         []string
+	excludedPackages []string
+	enabledServices  []string
+	disabledServices []string
+	kernelOptions    string
+	bootable         bool
+	defaultSize      uint64
+	assembler        func(uefi bool, size uint64) *osbuild.Assembler
+}
+
+type fedora31Arch struct {
```

Comment:
What is the reason for having 2 private "arch" struct here? (`fedora31Arch` and `arch`), seems to me that they have name in common, so the only difference is a pointer to the distro struct.

Reply:
This was to make the (already very large) patch a bit smaller. Would be a natural next cleanup. You are right, this is clearly not how it should be.

Reply:
We discussed this on the last PR: https://github.com/osbuild/osbuild-composer/pull/375#discussion_r396374144

Tom's idea is to do this in chunks.